src/java.base/share/classes/java/net/InetAddress.java

```diff
@@ -785,9 +785,12 @@ String getHostName(boolean check) {
     }
 
     /**
-     * Gets the fully qualified domain name for this IP address.
-     * Best effort method, meaning we may not be able to return
-     * the FQDN depending on the underlying system configuration.
+     * Gets the fully qualified domain name for this
+     * {@linkplain InetAddress#getAddress() IP address} using the system-wide
+     * {@linkplain InetAddressResolver resolver}. This is a best effort method,
+     * meaning we may not be able to return the fully qualified domain name; in
```

Comment:
I think you'll need to replace "we" with "it" here. It may also be useful to expand this sentence a bit, maybe with an example of why it may fail. 

I also wonder (and I'm in two minds on this) is if this method should include an impNote to say what it actually does.

Reply:
Hello Alan,

> I also wonder (and I'm in two minds on this) is if this method should include an impNote to say what it actually does.

I'm guessing that you are thinking of having the implNote state that a reverse name lookup will be performed by this method using the system-wide resolver. Right now, the `getHostName()` says this:

```
* <p>If this InetAddress was created with a host name,
     * this host name will be remembered and returned;
     * otherwise, a reverse name lookup will be performed
     * and the result will be returned based on the system
     * configured resolver. If a lookup of the name service
     * is required, call
     * {@link #getCanonicalHostName() getCanonicalHostName}.
```

I think we could update the `getCanonicalName()` to mention the reverse name lookup. However, I'm unsure if it should be an `implNote` or just formal API javadoc.

Reply:
I have now updated the PR to add a few more details on what this method does and why it might fail.

Reply:
> I'm guessing that you are thinking of having the implNote state that a reverse name lookup will be performed by this method using the system-wide resolver. 

I was thinking more about the caching and attempting the lookup only on first usage.  So my comment was wondering out loud if we should note this behavior in an implNote.

Reply:
> I was thinking more about the caching and attempting the lookup only on first usage.

If we do have to make a mention of that, then I think we would have to probably change the implementation to actually do the lookup only on first usage. Right now, the `canonicalHostName` field isn't `volatile` and there isn't any construct in place which will ensure that multiple concurrent calls to this method will ensure only one lookup happens. It's harmless in the current form, but if we add an `implNote` about this, then I think we would have to make it actually behave that way.